dev/sparktestsupport/modules.py

```diff
@@ -178,6 +178,10 @@ def __hash__(self):
     ],
     sbt_test_goals=[
         "avro/test",
+    ],
+    python_test_goals=[
+        # doctests
+        "pyspark.sql.avro.functions"
```

Comment:
@gaborgsomogyi, I think this isn't detected properly when there are only Avro related changes. Currently, the tests look being ran since `root` module (which means all modules) is detected.

Also, I think we should somehow put this into `pyspark-sql` module. Currently, this PR adds new module called `avro` into python testing script:

```
Running PySpark tests. Output is in /home/jenkins/workspace/SparkPullRequestBuilder/python/unit-tests.log
Will test against the following Python executables: ['python2.7', 'python3.4', 'pypy']
Will test the following Python modules: ['avro', 'pyspark-core', 'pyspark-ml', 'pyspark-mllib', 'pyspark-sql', 'pyspark-streaming']
Starting test(pypy): pyspark.sql.tests.test_catalog
```

Reply:
@HyukjinKwon 
> I think this isn't detected properly when there are only Avro related changes.

I've double checked the avro change only detection with the following:
```
determine_modules_for_files(["external/avro/foo"])
['avro']
```
and seems like it's working. Have you tested it differently?

> Also, I think we should somehow put this into `pyspark-sql` module.

You mean here adding the doc test into `pyspark-sql`s `python_test_goals`?

Reply:
Ah , I mean when the Python file `python/pyspark/sql/avro/functions.py` alone is fixed.

> You mean here adding the doc test into pyspark-sqls python_test_goals?

Yes, I should take a closer look (sorry I have been busy for the last week and I'm trying to clean up my notification queue) but I think that looks at least what kinesis does (which is the similar instance with avro's case)

Reply:
Can you add `avro` as dependency in `pyspark_sql`, add `pyspark.sql.avro.functions` into `python_test_goals`, and add this python file into `source_file_regexes`? I think that's going to be fine.

Reply:
One addition.
> add this python file into source_file_regexes

Since the related file starts with `python/pyspark/sql` which is already added to `source_file_regexes` I think it's not required.